wrappers/kn/src/nativeMain/kotlin/zxingcpp/BarcodeWriter.kt

```diff
@@ -0,0 +1,93 @@
+/*
+* Copyright 2024 ISNing
+*/
+// SPDX-License-Identifier: Apache-2.0
+
+package zxingcpp
+
+import cnames.structs.ZXing_CreatorOptions
+import cnames.structs.ZXing_WriterOptions
+import kotlinx.cinterop.*
+import zxingcpp.cinterop.*
+import kotlin.experimental.ExperimentalNativeApi
+import kotlin.native.ref.createCleaner
+
+// TODO: Remove this annotation when the API is stable
+@RequiresOptIn(level = RequiresOptIn.Level.ERROR, message = "The Writer API is experimental and may change in the future.")
+@Retention(AnnotationRetention.BINARY)
+annotation class ExperimentalWriterApi
+
+
+@ExperimentalWriterApi
+@OptIn(ExperimentalForeignApi::class)
+class BarcodeWriter : WriterOptions() {
```

Comment:
Looking at this, I'd like to question the utility of the `BarcodeWriter` class. Why not implement the `toSVG`/`toImage` members of `Barcode` directly calling the c-API and completely remove this class? With the above constructor idea the usecase could then look like
```Kotlin
val svg = Barcode("some test", BarcodeFormat.QRCode).toSVG();
```

Could you add something like this to the README.md? I'd like to always start from the client side usability and strive for a native feeling API. See also my current .NET and Rust README.

Reply:
I believe it's best to keep the BarcodeWriter class because it serves as a logical place to apply writer options. Moreover, this aligns with Java's style, even though it may not be the most convenient approach.

Regarding the toSVG and toImage methods, they have already been implemented in the code, as seen here:
https://github.com/zxing-cpp/zxing-cpp/pull/764/files#diff-108ceccf4e8d9a00451a5f5cca9c48695f005ddcc552c009530799a244f037a1R152-R157

By providing both methods for converting the barcode, users have the flexibility to choose the one that best suits their needs and preferences.

I will add the usage of `toSVG` and `toImage` later.

Reply:
The logical place to apply the options to is `WriterOptions`. Given my stance that `BarcodeWriter` is useless here, I thought about the reason for the existence of `BarcodeReader`. Why not have a `Barcode` constructor that takes an image? The answer is, the reader returns multiple objects, which a constructor can't.

What do you mean by "Java's style"?

Reply:
Because I understand that kotlin is more suited to java styles (a bunch of controllers, managers, blabla) because it influences the style of code that comes from java, and of course, in my experience, kotlin is more suited to functional programming, so I think both options are fine.

Reply:
Hmm, so why don't we change BarcodeReader to something like `ImageView.readBarcodes(opts)` or sth?
Dont you think that it's wierd there's `Reader` but not a `Writer`?

Reply:
I think maybe we're influenced by the design of original ZXing java